docs/upgrading.rst

```diff
@@ -149,3 +149,51 @@ Re-use of environments
 
     [testenv:b]
     deps = pytest<7
+
+
+Legacy CLI entry point may result in ambiguous usages
+-----------------------------------------------------
+
+``tox`` 4 introduced dedicated subcommands for various usages.
+However, when no recognized command is given the legacy entry point which imitates ``tox`` 3 is used.
+
+As a result, commands which were previously unambiguous may have become ambiguous.
+
+For example, consider the following tox config:
+
+.. code-block:: ini
+
+    [tox]
+    env_list = py39,py310
+
+    [testenv]
+    commands =
+        python -c 'print("hi")'
+
+    [testenv:list]
+    commands =
+        python -c 'print("a, b, c")'
+
+This defines an environment whose name matches a ``tox`` 4 command, ``list``.
+
+Under ``tox`` 3, ``tox -e list`` specified the ``list`` environment.
+
+Under ``tox`` 4, this usage parses as an invocation of ``tox list``.
+Therefore, attempting that same usage results in an error:
+
+.. code:: bash
+
+    $ tox -e list
+    ...
+    tox: error: unrecognized arguments: -e
+
+It can be run either with an explicit ``tox run`` or an explicit use of the
+legacy command:
+
+.. code:: bash
+
+    $ tox run -e list
```

Comment:
For what's worth this is my recommended way to run unambiguous command names too, so let's State that. Assuming legacy by default is just a backwards compatibility layer and might be removed with tox 5.

Reply:
Worth mentioning that you don't have to type run fully, the r letter acts as a shorter alias 👍

Reply:
Okay, I'll come back to you with something that states that more clearly, and we can wordsmith it more if necessary.

Reply:
In order to put this front-and-center, I decided to split the section up a little more.
First it states that you should rewrite any use of `tox -e foo` to `tox run -e foo`/`tox r -e foo` without trying to explain why. The second subsection focuses on the bad case of an environment matching a command name.

Hope you like it this way!